main/src/cgeo/geocaching/export/PersonalnoteExport.java

```diff
@@ -0,0 +1,99 @@
+import cgeo.geocaching.models.Geocache;
+import cgeo.geocaching.utils.AsyncTaskWithProgress;
+import cgeo.geocaching.utils.Log;
+
+import android.app.Activity;
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Exports personal notes.
+ *
+ */
+public class PersonalnoteExport extends AbstractExport {
+
+    private int persNotesCount = 0;
+
+    public PersonalnoteExport() {
+        super(R.string.export_persnotes);
+    }
+
+    @Override
+    public void export(@NonNull final List<Geocache> cachesList, @Nullable final Activity activity) {
+        final Geocache[] caches = cachesList.toArray(new Geocache[cachesList.size()]);
+        new ExportTask(activity).execute(caches);
+    }
+
+    private class ExportTask extends AsyncTaskWithProgress<Geocache, Boolean> {
+        private static final int STATUS_UPLOAD = -1;
+
+        /**
+         * Instantiates and configures the task for exporting personal notes.
+         */
+        ExportTask(@Nullable final Activity activity) {
+            super(activity, getProgressTitle(), CgeoApplication.getInstance().getString(R.string.export_persnotes_creating), true);
+        }
+
+        @Override
+        protected Boolean doInBackgroundInternal(final Geocache[] caches) {
+            // upload personal notes separately for each connector, so the file can be uploaded to the respective site afterwards
+            boolean success = true;
+            for (final IConnector connector : ConnectorFactory.getConnectors()) {
+                if (connector instanceof PersonalNoteCapability) {
+                    success &= uploadPersonalNotes((PersonalNoteCapability) connector, caches);
+                }
+            }
+            return success;
+        }
+
+        private boolean uploadPersonalNotes(final PersonalNoteCapability connector, final Geocache[] caches) {
+            try {
+                int i = 0;
+                for (final Geocache cache : caches) {
+                    if (ConnectorFactory.getConnector(cache).equals(connector) && StringUtils.isNotBlank(cache.getPersonalNote())) {
+                        connector.uploadPersonalNote(cache);
+                        persNotesCount++;
+                    }
+                    publishProgress(++i);
+                }
+            } catch (final Exception e) {
+                Log.e("PersonalnoteExport.ExportTask generation", e);
+                return false;
+            }
+            return true;
+        }
+
+        @Override
+        protected void onPostExecuteInternal(final Boolean result) {
+            if (activity != null) {
+                final Context nonNullActivity = activity;
+                if (result) {
+                    ActivityMixin.showToast(activity, nonNullActivity.getString(R.string.export_fieldnotes_upload_success));
```

Comment:
You should not reuse the existing string R.string.export_fieldnotes_upload_success. Even though it might simply say "Success" in English, translators of other languages may refer to field notes in there, since we gave it that id. If you want to use the same string in both cases, you basically have to introduce a new id "R.string.export_success" or similar.